library/kani_macros/src/lib.rs

```diff
@@ -49,3 +51,27 @@ pub fn proof(_attr: TokenStream, item: TokenStream) -> TokenStream {
     //     $item
     // )
 }
+
+#[cfg(not(kani))]
+#[proc_macro_attribute]
+pub fn unwind(_attr: TokenStream, _item: TokenStream) -> TokenStream {
+    // This code is never called by default as the config value is set to kani
+    TokenStream::new()
+}
+
+#[cfg(kani)]
+#[proc_macro_attribute]
+pub fn unwind(attr: TokenStream, item: TokenStream) -> TokenStream {
+    let mut result = TokenStream::new();
+
+    // Translate #[kani::unwind(arg)] to #[kanitool::unwind(8)]
+    let insert_string = "#[kanitool::unwind(".to_owned() + &attr.clone().to_string() + ")]";
+
+    // Add the string that looks like - #[kanitool::unwindvalue]
+    result.extend(insert_string.parse::<TokenStream>().unwrap());
+    // No mangle seems to be necessary as removing it prevents all the attributes in a lib from being read
+    result.extend("#[no_mangle]".parse::<TokenStream>().unwrap());
```

Comment:
Do we still need this?

Reply:
No, not anymore. let me update that.

Reply:
Update :  We do need this. Without it, all unwind annotations without `kani::proof` just get ignored and this prevents us from showing an error to the user. IMO, that would confuse the user as they would not know what the correct syntax for the annotation is. So, I think we should keep no_mangle on.

Reply:
No mangle has a side effect of making a function public and potentially causing name conflicts (E.g.: https://github.com/rust-lang/rust/issues/28179).

BTW, how did you test this behavior? Can you please make sure that the function you are annotating isn't dead code. Feel free to ping me offline.

Reply:
Talked offline. We'll remove this.